hello/lib/hello_web/plugs/locale.ex

```diff
@@ -0,0 +1,15 @@
+defmodule HelloWeb.Plugs.Locale do
+  import Plug.Conn
+
+  @locales ["en", "fr", "de"]
+
+  def init(default), do: default
+
+  def call(%Plug.Conn{params: %{"locale" => loc}} = conn, _default) when loc in @locales do
```

Comment:
@Vretamoza how does the call function work in a plug?

Reply:
It receives as second parameter what the init function returns and does what it is meant to do.

Reply:
🤔  could you please explain in deep?

Reply:
Plugs take requests and modify them as needed, in the init callback we initialize the arguments and pass them as second parameter in the call function which is the one that makes the required changes of the plug.